repose-aggregator/components/filters/ip-user-filter/src/test/scala/org/openrepose/filters/ipuser/IpUserFilterTest.scala

```diff
@@ -84,15 +84,98 @@ class IpUserFilterTest extends FunSpec with BeforeAndAfterEach with Matchers {
     }
   }
 
+  describe("setting the request x-pp-groups header") {
+    it("should use the value of the remote IP address of the request when no X-Forwarded-For header is included and group matches") {
+      servletRequest.setRemoteAddr("10.55.66.77")
+
+      ipUserFilter.doFilter(servletRequest, servletResponse, filterChain)
+
+      getPostFilterRequest.getHeader("x-pp-groups") shouldEqual "my-awesome-group" + DefaultQuality
+    }
+
+    it("should use the value of the X-Forwarded-For header when it is included") {
+      servletRequest.setRemoteAddr("10.1.2.3")
+      servletRequest.addHeader("X-Forwarded-For", "10.55.66.77")
+
+      ipUserFilter.doFilter(servletRequest, servletResponse, filterChain)
+
+      getPostFilterRequest.getHeader("x-pp-groups") shouldEqual "my-awesome-group" + DefaultQuality
+    }
+
+    it("group should be null when there are multiple non-split values in the X-Forwarded-For header - last match") {
+      servletRequest.setRemoteAddr("10.1.2.3")
+      servletRequest.addHeader("X-Forwarded-For", "10.1.1.1")
+      servletRequest.addHeader("X-Forwarded-For", "10.55.66.77")
+
+      ipUserFilter.doFilter(servletRequest, servletResponse, filterChain)
+
+      getPostFilterRequest.getHeader("x-pp-groups") should be (null)
```

Comment:
why would this be null?

Reply:
I see you changed the config being tested.

Reply:
That header doesn't populate if nothing matches.